src/core/packages/saved-objects/migration-server-internal/src/common/utils/logs.ts

```diff
@@ -69,3 +66,17 @@ export const logActionResponse = (
 ) => {
   logger.debug(logMessagePrefix + `${state.controlState} RESPONSE`);
 };
+
+export function logMigrationFailureDetails({
+  logger,
+  index,
+  cstDiag,
+}: {
+  logger: Logger;
+  index: string;
+  cstDiag: ControlStateTransitionDiag;
+}): void {
+  logger
+    .get('control-state-transitions')
+    .error(`[${index}] Migration failed after transitioning through: ${cstDiag.prettyPrint()}`);
```

Comment:
Trying to preserve debuggability here, assuming we only care about state transitions when something goes wrong so capture same transition info in a single error message.